### PR TITLE
Uxd 835 fix takeover types

### PR DESCRIPTION
**Pull request by allison-c**
### Purpose 🚀

Trying to fix takeover types and docs generation. The problem is lots of components are reusing the proptypes definition from another component/package, react docgen doesn't have the ability to read from the original source. This is another good reason to move away from reusing propTypes. 

Another problem is the default type of each prop was `-` before, not sure why we set that, but definitely not a valid value in ts. So I changed it to `any`, to avoid parsing error.

### Notes ✏️
_details of code change / secondary purposes of this PR_

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_ (if the package you are breaking is a dependency in other packages, be sure to add a commit with a "BREAKING CHANGE" message to those other packages as well)
- [ ] MINOR (backward compatible) change to _these packages_
- [x] PATCH (bug fix) change to `@paprika/takeover`
### Storybook 📕
http://storybooks.highbond-s3.com/paprika/your-branch-name

### Screenshots 📸
_optional but highly recommended_

### References 🔗
_relevant Jira ticket / GitHub issues_

### CODE REVIEW

**Comment by nahumzs** [comment]
`reason to move away from reusing propTypes` <---- 110% agree with this, repeat yourself sometimes is better than DRY

File: packages/Takeover/README.md [hunk 1]
```diff
@@ -32,6 +32,23 @@ npm install @paprika/takeover
 | onClose      | func   | false    | () => {}  | Callback triggered when the takeover needs to be close |
 | zIndex       | number | false    | zValue(5) | The z-index of the Takeover content                    |
 
+### Takeover.Header
```
Comment (nahumzs) [inline packages/Takeover/README.md:35]: 👏

File: packages/Takeover/src/Takeover.js [hunk 2]
```diff
@@ -39,32 +40,33 @@ const defaultProps = {
...
     ...(focusLockProps || {}),
   };
 
-  const takeoverOverlay = overlayExtracted || <TakeoverOverlay />;
   const overlayProps = {
     "data-pka-anchor": "takeover.overlay",
     hasBackdrop: false,
     focusLockOptions,
-    ...takeoverOverlay.props,
+    ...(overlayExtracted ? overlayExtracted.props : {}),
```
Comment (nahumzs) [inline packages/Takeover/src/Takeover.js:69]: why is this need it? `...undefined` would no spread anything still isn't it?

File: packages/Takeover/src/Takeover.js [hunk 3]
```diff
@@ -84,25 +86,29 @@ export default function Takeover(props) {
...
-      zIndex={zIndex}
-    >
-      {headerExtracted && <sc.Header {...headerExtracted.props} onClose={onClose} />}
-      {contentExtracted && (
-        <sc.Content role="region" tabIndex="0">
-          {contentExtracted}
-        </sc.Content>
+  return (
+    <OriginalOverlay {...overlayProps}>
+      {state => (
```
Comment (nahumzs) [inline packages/Takeover/src/Takeover.js:91]: why is this turn into a render function? I'm not understanding the context.

Comment (allison-c) [reply packages/Takeover/src/Takeover.js:91]: it was a render function before as well, https://github.com/acl-services/paprika/pull/934/files/97c7748ee1ea3f36898a5fddc36ab2519afbff15#diff-7f8e871aef4f5eab982513c0ec898d0387c02987deb2af97b5d961413ab0fea4L87 look at the children parameter was a function. This is a requirement from `@paprika/overlay` component

**Review by nahumzs — APPROVED** [review]

**Pull request merged**